doc/HACKING.md

```diff
@@ -0,0 +1,27 @@
+# Hacking on the kata-operator
```

Comment:
I thinking Hacking is cool in the content of the doc, but for the title (also the first doc we add :D) maybe `local-development` / `development` or something along would fit more? WDYT?

Reply:
I thought the name was common in related projects. machine-config-operator has the same file, opencontaimer/image-spec, cluster-etc-operator all have a HACKING.md. I wanted not to deviate from that. But that's the only reason. I have no strong opinion either way, so I'll just name it DEVELOPMENT. Ok?

Reply:
Sure, the reason I commented is that other OS projects have a dedicated folder usually called `hack` that contains scripts for many things. But it is up to you, no strong opinions on my side as well :)

Reply:
ah, I did not get that from your comment. Sure, I think we add a helper script in hack/ to build a custom payload but I'd prefer to do that in another PR.

Reply:
It looks good as it is now :)